### PR TITLE
Weekly CLI backmerge

### PR DESCRIPTION
**Pull request by technophile-04**
## Description

- App router migration (#535)
- Update hardhat package plugins and add hardhat-verify (#637)

Tried to peacefully resolve all merge conflicts and also tested it nicely, works great for me 🙌 

but would love if others too could just give it nice test and skim through file changed once 🙌

NOTE: We shouldn't squash & merge it instead do a normal merge

### CODE REVIEW

File: templates/base/README.md [hunk 1]
```diff
@@ -114,6 +114,14 @@ You can verify your smart contract on Etherscan by running:
 yarn verify --network network_name
 ```
 
+eg: `yarn verify --network sepolia`
+
+This uses [etherscan-verify from hardhat-deploy](https://www.npmjs.com/package/hardhat-deploy#4-hardhat-etherscan-verify) to verify all the deployed contracts.
+
+You can alternatively use [hardhat-verify](https://hardhat.org/hardhat-runner/plugins/nomicfoundation-hardhat-verify) to verify your contracts by running `yarn hardhat-verify --network network_name contract_address`
```
Comment (rin-st) [inline templates/base/README.md:121]: There should be also be constructor arguments see https://arc.net/l/quote/fqupvdti

**Review by rin-st — APPROVED** [review]
Left comment regarding verify constructor args. Other than that looks great to me!

File: templates/base/README.md (hunk 1 above)
Comment (technophile-04) [reply templates/base/README.md]: Ohh thanks Rinat! update at [71be8e7](https://github.com/scaffold-eth/scaffold-eth-2/pull/685/commits/71be8e7379b10b240d06ca7e7393ff7e2a856505) ....Also just learned about this Arc browser quote link feature love it !! :)

Thanks 🙌

Comment (rin-st) [reply templates/base/README.md]: Great, thanks!

File: templates/base/packages/nextjs/app/page.tsx
Comment (Pabl0cks) [inline templates/base/packages/nextjs/app/page.tsx:1]: We need to update page url, from this:  
<code className="italic bg-base-300 text-base font-bold max-w-full break-words break-all inline-block">
              packages/nextjs/pages/index.tsx
</code>

To this:  
<code className="italic bg-base-300 text-base font-bold max-w-full break-words break-all inline-block">
              packages/nextjs/app/page.tsx
</code>

Will create a PR to fix it in SE-2 repo.

Comment (Pabl0cks) [reply templates/base/packages/nextjs/app/page.tsx:1]: Just pushed the update

**Review by Pabl0cks — APPROVED** [review]
Looks good to me! Pushed a couple of minor tweaks 🙌

Great job Shiv!! 🔥

**Pull request merged**